### PR TITLE
TPCQC: Tracks Task Updated with maps

### PR DESCRIPTION
**Pull request by sbhawani**

### CODE REVIEW

File: Detectors/TPC/qc/include/TPCQC/Tracks.h [hunk 1]
```diff
@@ -66,19 +69,38 @@ class Tracks
...
   const std::vector<TH1F>& getHistogramRatios1D() const { return mHistRatio1D; }
 
+  /// get ratios of 1D histograms
+  std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getmMapHist() { return mMapHist; }
+  const std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getmMapHist() const { return mMapHist; }
+
  private:
+  float mCutAbsEta;  // Eta cut
+  int mCutnCls;      // minimum N clusters
+  float mCutdEdxTot; // dEdxTot min value
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/Tracks.h]:
```suggestion
  float mCutAbsEta;  // Eta cut
  int mCutMinnCls;      // minimum N clusters
  float mCutMindEdxTot; // dEdxTot min value
```

File: Detectors/TPC/qc/include/TPCQC/Tracks.h [hunk 2]
```diff
@@ -66,19 +69,38 @@ class Tracks
...
   const std::vector<TH2F>& getHistograms2D() const { return mHist2D; }
 
+  // Just for backward compatibility with crrent QC, temporary, will be removed in the next PR
   /// get ratios of 1D histograms
   std::vector<TH1F>& getHistogramRatios1D() { return mHistRatio1D; }
   const std::vector<TH1F>& getHistogramRatios1D() const { return mHistRatio1D; }
 
+  /// get ratios of 1D histograms
+  std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getmMapHist() { return mMapHist; }
+  const std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getmMapHist() const { return mMapHist; }
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/Tracks.h]:
```suggestion
  std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getMapHist() { return mMapHist; }
  const std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getMapHist() const { return mMapHist; }
```

File: Detectors/TPC/qc/src/Tracks.cxx [hunk 3]
```diff
@@ -27,62 +27,64 @@
 ClassImp(o2::tpc::qc::Tracks);
 
 using namespace o2::tpc::qc;
+const std::vector<std::string_view> histVecNames{"hNClustersBeforeCuts", "hNClustersAfterCuts", "hEta",
+                                                 "hPhiAside", "hPhiCside", "hPt", "hSign", "hEtaNeg", "hEtaPos", "hPhiAsideNeg", "hPhiAsidePos", "hPhiCsideNeg", "hPhiCsidePos",
+                                                 "hPtNeg", "hPtPos", "hEtaBeforeCuts", "hPtBeforeCuts", "hQOverPt", "hPhiBothSides", "h2DNClustersEta", "h2DNClustersPhiAside", "h2DNClustersPhiCside",
+                                                 "h2DNClustersPt", "h2DEtaPhi", "h2DEtaPhiNeg", "h2DEtaPhiPos", "h2DNClustersEtaBeforeCuts", "h2DQOverPtPhiAside", "h2DQOverPtPhiAside",
+                                                 "h2DQOverPtPhiCside", "hEtaRatio", "hPhiAsideRatio", "hPhiCsideRatio", "hPtRatio"};
```
Comment (wiechula) [inline Detectors/TPC/qc/src/Tracks.cxx]: is this used somewhere?

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Detectors/TPC/qc/src/Tracks.cxx (hunk 3 above)
Comment (sbhawani) [reply Detectors/TPC/qc/src/Tracks.cxx]: right,  this is not required I removed it

File: Detectors/TPC/qc/include/TPCQC/Tracks.h (hunk 1 above)
Comment (sbhawani) [reply Detectors/TPC/qc/include/TPCQC/Tracks.h]: done

**Comment by sbhawani** [comment]
This PR is ready and can be merged.  @wiechula and @davidrohr can you please approve ? thanks

File: Detectors/TPC/qc/include/TPCQC/Tracks.h [hunk 4]
```diff
@@ -66,19 +69,38 @@ class Tracks
...
   const std::vector<TH1F>& getHistogramRatios1D() const { return mHistRatio1D; }
 
+  /// get ratios of 1D histograms
+  std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getMapHist() { return mMapHist; }
+  const std::unordered_map<std::string_view, std::unique_ptr<TH1>>& getMapHist() const { return mMapHist; }
+
  private:
+  float mCutAbsEta;     // Eta cut
+  int mCutMinnCls;      // minimum N clusters
+  float mCutMindEdxTot; // dEdxTot min value
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/Tracks.h]: I think default values would be good. Perhaps:
```suggestion
  float mCutAbsEta = 1.f;     // Eta cut
  int mCutMinnCls = 60;      // minimum N clusters
  float mCutMindEdxTot = 20.f; // dEdxTot min value
```

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Detectors/TPC/qc/include/TPCQC/Tracks.h (hunk 4 above)
Comment (sbhawani) [reply Detectors/TPC/qc/include/TPCQC/Tracks.h]: yes it is better.. included in

**Comment by sbhawani** [comment]
@wiechula @davidrohr @shahor02, can this PR be merged? its ready. thanks

**Review by wiechula — DISMISSED** [review]

**Review by wiechula — APPROVED** [review]

**Comment by tklemenz** [comment]
The problem seems unrelated. Can this be merged?

**Pull request merged**